controllers/managedocs_controller.go

```diff
@@ -136,3 +145,34 @@ func (r *ManagedOCSReconciler) setDesiredStorageCluster(
 
 	return nil
 }
+
+func (r *ManagedOCSReconciler) updateComponents(req ctrl.Request) ([]v1.ManagedOCSComponent, error) {
+	var components []v1.ManagedOCSComponent
+
+	// Checking the StorageCluster component.
+	component := v1.ManagedOCSComponent{
+		Name: "StorageCluster",
+	}
+
+	var scList ocsv1.StorageClusterList
+
+	if err := r.List(r.ctx, &scList); err != nil {
+		r.Log.Error(err, "error listing StorageCluster")
+		return nil, err
+	}
+
+	if len(scList.Items) < 1 {
+		component.Status = v1.ComponentNone
+	} else {
+		// Assuming only one StorageCluster in namespace.
+		if scList.Items[0].Status.Phase == "Ready" {
```

Comment:
It is better to search for the item in a list by name not by location. future changes to ocs might result in lists with more than 1 item